### DISCUSSION TITLE
Meu codigo não está executando

### DISCUSSION THREAD

**Discussion by Squ3let0n** [frontendbr/forum#2154] [Perguntas e Respostas]
const students = [
{  
    name: 'Will',
    grade: 5
},
{
    name: 'Rafaela',
    grade: 7
}
];
function getStudent(studentsList) {
    return studentsList.filter(student => student.grade >= 5);
}
console.log('Alunos aprovados:');
console.log(getAprovedStudents(studentsList));

console.log('\nLista de alunos:');
console.log(studentsList);

----------------------------------------------------------------------------------------------------------------------------------------------
Alguem sabe oq pode ser ?

**Comment by ninetails** [comment]
Bom, vamos separar o seu código:

```js
const students = [
  {
    name: 'Will',
    grade: 5
  },
  {
    name: 'Rafaela',
    grade: 7
  }
];
```
Nestas linhas você está instanciando um objeto literal chamado `students`.

---

```js
function getStudent(studentsList) {
  return studentsList.filter(student => student.grade >= 5);
}
```

Nestas linhas você criou uma função chamada `getStudent` que recebe uma `studentsList` e que utiliza dentro dela no formato definido no `students` (mas vc só falou que recebe uma variável que tem essa especificação).

---

`console.log` provavelmente vc já viu que é só uma função para _printar_, certo?

Nas utilizações dentro dos seguintes `console.log`:

```js
console.log(getAprovedStudents(studentsList));
console.log(studentsList);
```

você está chamando uma variável chamada `studentsList`, porém você não definiu nenhuma variável chamada `studentsList`. O que você definiu anteriormente com este nome é o nome do argumento de uma função que existirá somente dentro da chamada dela, ou seja, somente dentro de `getStudent`. Fora desta função, não tem nenhum `studentsList` definido.

O que você definiu anteriormente foi um objeto literal chamado `students`.

---

Bom, daí dá pra figurar, certo?

Caso ainda não, busque sobre **Escopos e variáveis** no Google pois a maioria dos programas de computador funcionam assim.

**Reply by Squ3let0n** [reply]
Entendi, muito obrigado
